companion/src/radiodata.cpp

```diff
@@ -39,61 +39,97 @@ RawSource RawSource::convert(Board::Type before, Board::Type after)
     }
   }
 
-  // No S3, LS and RS on X7 board
-  if (IS_TARANIS_X7(after)) {
-    if (type == SOURCE_TYPE_STICK && index >= 6) {
-      index = 5;
-    }
-  }
-
   // SWI to SWR don't exist on !X9E board
   if (!IS_TARANIS_X9E(after) && IS_TARANIS_X9E(before)) {
     if (type == SOURCE_TYPE_SWITCH && index >= 8) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)